lib/elixir/lib/code.ex

```diff
@@ -659,6 +663,33 @@ defmodule Code do
   The opposite of converting a string to its quoted form is
   `Macro.to_string/2`, which converts a quoted form to a string/binary
   representation.
+
+  ## The `:static_atom_encoder` function
+
+  When `static_atom_encoder: &my_encoder/2` is passed as an argument,
+  `my_encoder/2` is called every time the tokenizer needs to create a
+  'static' atom. Static atoms are atoms in the AST that function as
+  aliases, remote calls, local calls, variable names, regular atoms
+  and keyword lists.
+
+  The encoder function will receive the atom name (as a binary) and a
+  keyword list with the current file, line and column. It must return
+  `{:ok, token :: term} | {:error, reason :: binary}`.
+
+  The encoder function is supposed to create an atom from the given
+  string. It is required to return either `{:ok, term}`, where term is
+  an atom. It is possible to return something else than an atom,
+  however, in that case the AST is no longer "valid" in that it cannot
+  be used to compile or evaluate Elixir code. A use case for this is
+  if you want to use the Elixir parser in a user-facing situation, but
+  you don't want to exhaust the atom table.
+
+  When the encoder fails for some reason, it must return `{:error,
+  reason :: binary}`. The atom encoder is not called for *all* atoms
+  that are present in the AST: it won't be invoked for operators,
+  syntax keywords (fn, do, etc), and neither for atoms containing
+  interpolations.
```

Comment:
This is an important note, so I would write it in a more formal way:

```
The atom encoder is not called for *all* atoms that are present in
the AST. It won't be invoked for the following atoms:

  * operators (`:+`, `:-`, and so on)

  * syntax keywords (`fn`, `do`, and so on)

  * atoms containing interpolation (`:"#{1 + 1} is two"`)
```

Wdyt?

Reply:
Yep, looks good, done in f9ae8e7